internal/security/credentials_test.go

```diff
@@ -39,10 +39,10 @@ type getSecretsTestObj struct {
 	resetSecretsCache bool
 }
 
-var getSecretsTestData []getSecretsTestObj
+type secretTestData []getSecretsTestObj
 
-func TestMain(m *testing.M) {
-	getSecretsTestData = []getSecretsTestObj{
+func getSecretsTestData() secretTestData {
+	return []getSecretsTestObj{
```

Comment:
self-note:  this change on the test is to resolve the potential threading issue when running  `make test` or running in package testing mode inside VS code.  

Sometimes, you will get the error like this: 
```
=== RUN   TestGetInsecureSecrets/Two_valid_keys_(key1_cached,_key2_not_cached)
--- FAIL: TestGetInsecureSecrets (0.00s)
    --- PASS: TestGetInsecureSecrets/Empty_path (0.00s)
    --- PASS: TestGetInsecureSecrets/Empty_path,_keys_list_empty (0.00s)
    --- PASS: TestGetInsecureSecrets/Fake_path (0.00s)
    --- PASS: TestGetInsecureSecrets/Empty_keys (0.00s)
    --- PASS: TestGetInsecureSecrets/One_valid_key,_one_empty_key (0.00s)
    --- PASS: TestGetInsecureSecrets/One_valid_key_one_not_found_key (0.00s)
    --- PASS: TestGetInsecureSecrets/Not_found_key (0.00s)
    --- PASS: TestGetInsecureSecrets/Two_missing_keys (0.00s)
    --- PASS: TestGetInsecureSecrets/Valid_key (0.00s)
    --- PASS: TestGetInsecureSecrets/Two_valid_keys (0.00s)
    --- FAIL: TestGetInsecureSecrets/Valid_key_(key1_not_cached) (0.00s)
        /home/jim/go/src/github.com/edgexfoundry/app-functions-sdk-go/internal/security/credentials_test.go:187: 
            	Error Trace:	credentials_test.go:187
            	Error:      	Not equal: 
            	            	expected: map[string]string{"key1":"value1", "key2":"value2"}
            	            	actual  : map[string]string{"key1":"value1"}
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,4 +1,3 @@
            	            	-(map[string]string) (len=2) {
            	            	- (string) (len=4) "key1": (string) (len=6) "value1",
            	            	- (string) (len=4) "key2": (string) (len=6) "value2"
            	            	+(map[string]string) (len=1) {
            	            	+ (string) (len=4) "key1": (string) (len=6) "value1"
            	            	 }
            	Test:       	TestGetInsecureSecrets/Valid_key_(key1_not_cached)
    --- PASS: TestGetInsecureSecrets/One_valid_key_(key1_already_cached) (0.00s)
    --- PASS: TestGetInsecureSecrets/Two_valid_keys_(key1_cached,_key2_not_cached) (0.00s)
=== RUN   TestNewSecretProvider
--- PASS: TestNewSecretProvider (0.00s)
```